crates/storage/src/connection/event.rs

```diff
@@ -467,28 +550,63 @@ impl Transaction<'_> {
         })
     }
 
-    // TODO:
-    // Implement once [`AggregateBloom`] table is added.
-    fn _running_bloom_aggregate(&self) -> Result<Option<AggregateBloom>, anyhow::Error> {
-        // Fetch running aggregate from DB
-        unimplemented!()
+    #[cfg(feature = "aggregate_bloom")]
+    fn load_aggregate_bloom(
+        &self,
+        block_number: BlockNumber,
+    ) -> anyhow::Result<Option<AggregateBloom>> {
+        #[rustfmt::skip]
+        let mut select_stmt = self.inner().prepare_cached(
+            "SELECT from_block, to_block, bloom FROM starknet_event_filters_aggregate \
+            WHERE from_block <= ? AND to_block >= ?",
+        )?;
+
+        let aggregate = select_stmt
+            .query_row(params![&block_number, &block_number], |row| {
+                let from_block: u64 = row.get(0)?;
+                let to_block: u64 = row.get(1)?;
+                let compressed_bitmap: Vec<u8> = row.get(2)?;
+
+                Ok((from_block, to_block, compressed_bitmap))
+            })
+            .optional()
+            .context("Querying running bloom aggregate")?
+            .map(|(from_block, to_block, compressed_bitmap)| {
+                AggregateBloom::from_existing_compressed(from_block, to_block, compressed_bitmap)
+            });
+
+        Ok(aggregate)
     }
 
-    fn _load_bloom_range(
+    #[cfg(feature = "aggregate_bloom")]
+    fn load_aggregate_bloom_range(
         &self,
-        _from_block: BlockNumber,
-        _to_block: BlockNumber,
+        start_block: BlockNumber,
+        end_block: BlockNumber,
     ) -> anyhow::Result<Vec<AggregateBloom>> {
-        // Should be something like:
-        // (from_block..to_block)
-        //     .chunks(AggregateBloom::BLOCK_RANGE_LEN)
-        //     .iter()
-        //     .enumerate()
-        //     .map(|(i, _)| {
-        //         // load from DB where ID is i
-        //      })
-        //      .collect()
-        unimplemented!()
+        #[rustfmt::skip]
+        let mut stmt = self.inner().prepare_cached(
+            "SELECT from_block, to_block, bloom FROM starknet_event_filters_aggregate \
+            WHERE from_block <= ? AND to_block >= ? \
+            ORDER BY from_block",
+        )?;
+
```

Comment:
As far as I understand this loads a _single_ aggregate filter that contains the _full_ range of (start_block, end_block). How will this work if the (start_block, end_block) range comprises multiple aggregate filters?

Reply:
Indeed. Could we maybe make this more explicit in the statement itself by using [named params](https://www.sqlite.org/lang_expr.html#varparam)?